### ISSUE TITLE
fatal error: concurrent map writes

### ISSUE THREAD

**Issue by stevie-** [infracost/infracost#1202]
When we run infracost in our github action workflow we often see a fatal error.

Versions:
```
Download action repository 'infracost/actions@v1' (SHA:dc9f4f16929775a70b3b5164c007f46695f53f37)
Download action repository 'infracost/actions@comment-tag' (SHA:524f39cf4f16f5ebe042fbf28298f7183d7c9ca1)
###
Notice: Setup Infracost CLI version 0.9.15
```

```
Run infracost breakdown --config-file=/tmp/infracost-config.yml --format=json --out-file=/tmp/infracost.json
  infracost breakdown --config-file=/tmp/infracost-config.yml --format=json --out-file=/tmp/infracost.json
  shell: /bin/bash -e {0}
  env:
    AWS_ACCOUNT: central
    AWS_ROLE_TO_ASSUME: REDACTED
    AWS_DEFAULT_REGION: eu-central-1
    AWS_REGION: eu-central-1
    AWS_ACCESS_KEY_ID: ***
    AWS_SECRET_ACCESS_KEY: ***
    AWS_SESSION_TOKEN: ***
    TERRAFORM_VERSION: 1.0.10
    TERRAGRUNT_VERSION: 0.35.13
    TF_PLUGIN_CACHE_DIR: /home/runner/.terraform.d/plugin-cache
    TERRAGRUNT_DOWNLOAD: /home/runner/.terragrunt-cache//2021-12-15
    infracost_configfile: /tmp/infracost-config.yml
    infracost_comment_tag: Validate Terraform Code with central account
    INFRACOST_GITHUB_ACTION: true
    INFRACOST_SKIP_UPDATE_CHECK: true
    INFRACOST_VCS_REPOSITORY_URL: https://github.com/REDACTED
    INFRACOST_VCS_PULL_REQUEST_URL: https://github.com/REDACTED/pull/392
    INFRACOST_LOG_LEVEL: info
fatal error: concurrent map writes

goroutine 22 [running]:
runtime.throw({0x142f960, 0xc0001048d0})
	/opt/hostedtoolcache/go/1.17.4/x64/src/runtime/panic.go:1198 +0x71 fp=0xc000104878 sp=0xc000104848 pc=0x433631
runtime.mapassign_faststr(0xc00009b3fb, 0x15d9494, {0xc00004c090, 0x4})
	/opt/hostedtoolcache/go/1.17.4/x64/src/runtime/map_faststr.go:211 +0x39c fp=0xc0001048e0 sp=0xc000104878 pc=0x412a9c
github.com/sanathkr/go-yaml.registerCustomTagUnmarshaler(...)
	/home/runner/go/pkg/mod/github.com/sanathkr/go-yaml@v0.0.0-20170819195128-ed9d249f429b/decode.go:213
github.com/sanathkr/go-yaml.RegisterTagUnmarshaler(...)
	/home/runner/go/pkg/mod/github.com/sanathkr/go-yaml@v0.0.0-20170819195128-ed9d249f429b/yaml.go:169
github.com/awslabs/goformation/v4/intrinsics.registerTagMarshallers()
	/home/runner/go/pkg/mod/github.com/awslabs/goformation/v4@v4.19.5/intrinsics/tags.go:39 +0x9a fp=0xc000104938 sp=0xc0001048e0 pc=0x74c45a
github.com/awslabs/goformation/v4/intrinsics.ProcessYAML({0xc00009a000, 0x13fb, 0x13fc}, 0xc000104978)
	/home/runner/go/pkg/mod/github.com/awslabs/goformation/v4@v4.19.5/intrinsics/intrinsics.go:64 +0x31 fp=0xc000104988 sp=0xc000104938 pc=0x74a691
github.com/awslabs/goformation/v4.ParseYAMLWithOptions({0xc00009a000, 0x0, 0xc0002fc000}, 0x0)
	/home/runner/go/pkg/mod/github.com/awslabs/goformation/v4@v4.19.5/goformation.go:47 +0x1e fp=0xc0001049b8 sp=0xc000104988 pc=0xeb4d1e
github.com/awslabs/goformation/v4.OpenWithOptions({0xc0002fc000, 0xc0002fc000}, 0x1604040)
	/home/runner/go/pkg/mod/github.com/awslabs/goformation/v4@v4.19.5/goformation.go:35 +0x90 fp=0xc000104a10 sp=0xc0001049b8 pc=0xeb4c90
github.com/awslabs/goformation/v4.Open(...)
	/home/runner/go/pkg/mod/github.com/awslabs/goformation/v4@v4.19.5/goformation.go:17
github.com/infracost/infracost/internal/providers.isCloudFormationTemplate({0xc0002fc000, 0x0})
	/home/runner/work/infracost/infracost/internal/providers/detect.go:159 +0x25 fp=0xc000104a38 sp=0xc000104a10 pc=0xeb7705
github.com/infracost/infracost/internal/providers.Detect(0xc000094000)
	/home/runner/work/infracost/infracost/internal/providers/detect.go:26 +0x77 fp=0xc000104aa8 sp=0xc000104a38 pc=0xeb6a57
main.runProjectConfig(0xc0002ee280, 0xc0002a53b0, 0xc000094000, 0xc00019ee10)
	/home/runner/work/infracost/infracost/cmd/infracost/run.go:189 +0xd2 fp=0xc000104ee8 sp=0xc000104aa8 pc=0xebef52
main.runMain.func1()
	/home/runner/work/infracost/infracost/cmd/infracost/run.go:84 +0x159 fp=0xc000104f78 sp=0xc000104ee8 pc=0xebee19
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x67 fp=0xc000104fe0 sp=0xc000104f78 pc=0xeaf4a7
runtime.goexit()
	/opt/hostedtoolcache/go/1.17.4/x64/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc000104fe8 sp=0xc000104fe0 pc=0x463901
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x92

goroutine 1 [semacquire]:
sync.runtime_Semacquire(0xc0002df880)
	/opt/hostedtoolcache/go/1.17.4/x64/src/runtime/sema.go:56 +0x25
sync.(*WaitGroup).Wait(0xc0002e2b38)
	/opt/hostedtoolcache/go/1.17.4/x64/src/sync/waitgroup.go:130 +0x71
golang.org/x/sync/errgroup.(*Group).Wait(0xc00031ac60)
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:40 +0x27
main.runMain(0xc0002ee280, 0xc0002a53b0)
	/home/runner/work/infracost/infracost/cmd/infracost/run.go:101 +0x492
main.breakdownCmd.func1(0xc0002ee280, {0x14197b0, 0x3, 0x3})
	/home/runner/work/infracost/infracost/cmd/infracost/breakdown.go:42 +0x15c
github.com/spf13/cobra.(*Command).execute(0xc0002ee280, {0xc0002a5980, 0x3, 0x3})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002ad180)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main()
	/home/runner/work/infracost/infracost/cmd/infracost/main.go:57 +0x116

goroutine 18 [chan send]:
main.startUpdateCheck.func1()
	/home/runner/work/infracost/infracost/cmd/infracost/main.go:144 +0x9e
created by main.startUpdateCheck
	/home/runner/work/infracost/infracost/cmd/infracost/main.go:139 +0x77

goroutine 19 [runnable]:
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x92

goroutine 20 [runnable]:
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x92

goroutine 21 [runnable]:
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x92

goroutine 23 [runnable]:
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x92

goroutine 24 [runnable]:
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x92

goroutine 25 [runnable]:
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x92

goroutine 26 [runnable]:
internal/poll.runtime_pollOpen(0x3)
	/opt/hostedtoolcache/go/1.17.4/x64/src/runtime/netpoll.go:173 +0x146
internal/poll.(*pollDesc).init(0xc0002e2c78, 0x80001)
	/opt/hostedtoolcache/go/1.17.4/x64/src/internal/poll/fd_poll_runtime.go:40 +0x4d
internal/poll.(*FD).Init(0xc0002e2c60, {0x1419754, 0xffffffffffffff9c}, 0x0)
	/opt/hostedtoolcache/go/1.17.4/x64/src/internal/poll/fd_unix.go:63 +0x46
os.newFile(0x3, {0xc0001abb00, 0x7b}, 0x1)
	/opt/hostedtoolcache/go/1.17.4/x64/src/os/file_unix.go:172 +0x113
os.openFileNolog({0xc0001abb00, 0x13b8620}, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.17.4/x64/src/os/file_unix.go:241 +0x170
os.OpenFile({0xc0001abb00, 0x7b}, 0x0, 0x3ca1)
	/opt/hostedtoolcache/go/1.17.4/x64/src/os/file.go:338 +0x45
os.Open(...)
	/opt/hostedtoolcache/go/1.17.4/x64/src/os/file.go:318
os.ReadFile({0xc0001abb00, 0xc0001abb00})
	/opt/hostedtoolcache/go/1.17.4/x64/src/os/file.go:670 +0x6a
github.com/infracost/infracost/internal/providers.isTerraformPlanJSON({0xc0001abb00, 0x7b})
	/home/runner/work/infracost/infracost/internal/providers/detect.go:58 +0x2a
github.com/infracost/infracost/internal/providers.Detect(0xc00031ae10)
	/home/runner/work/infracost/infracost/internal/providers/detect.go:30 +0xf5
main.runProjectConfig(0xc0002ee280, 0xc0002a53b0, 0xc00031ae10, 0xc00019eea0)
	/home/runner/work/infracost/infracost/cmd/infracost/run.go:189 +0xd2
main.runMain.func1()
	/home/runner/work/infracost/infracost/cmd/infracost/run.go:84 +0x159
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x67
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x92
Error: Process completed with exit code 2.
```

**Comment by aliscott** [comment]
Looks related to https://github.com/awslabs/goformation/issues/363.

If you run with the env variable `INFRACOST_PARALLELISM: 1` it should work okay just now, and we'll look at getting a fix out on our end.

**Comment by aliscott** [comment]
A fix for this has now been merged in https://github.com/infracost/infracost/pull/1206 and will be released in v0.9.16 which will be released in about a week.

**Comment by vdmgolub** [comment]
@stevie- this is live in v0.9.16.
